src/Checkout/Helpers/ReserveStock.php

```diff
@@ -36,6 +36,24 @@ protected function is_enabled() {
 		return $this->enabled;
 	}
 
+	/**
+	 * Returns the amount of minutes to use for reserving stock stock.
+	 *
+	 * @param  integer $minutes amount of minutes to reserve stock.
+	 * @return integer $minutes filtered minutes.
+	 */
+	public function reservation_minutes( $minutes = 0 ) {
+
+		$minutes = $minutes ? $minutes : (int) get_option( 'woocommerce_hold_stock_minutes', 60 );
+
+		/**
+		 * Filter can also be false to bypass stock reservation later.
+		 */
+		$minutes = apply_filters( 'woocommerce_reserve_stock_minutes', $minutes );
```

Comment:
I don't know that we should allow developers to universally alter the length of time that stock is held for. If we're intending that this utility class be used by extension developers it might be problematic if a plugin that is attempting to alter length of time for one stock reservation area alters it for all of them.

Reply:
I do think it valuable to find a way that we can allow additional products to be added to the current reservation time. E.g when a customer adds a product to their cart the stock is reserved for X minutes. If customers add a second product we need the flexibility to be able to add that product to the original reservation time.

Reply:
In that specific case (since we're now also going to pass the customer ID), do you think the existing `woocommerce_query_for_reserved_stock` filter could be used for this purpose? This gives you the information about the product, the order ID, and the customer ID, which could then be used to alter the expiration in the query?